solidjs-tailwind/src/pages/Signin.jsx

```diff
@@ -0,0 +1,40 @@
+import { createEffect, createResource } from 'solid-js';
+import { useAuth } from '../auth';
+import { useNavigate } from '@solidjs/router';
+
+const fetchToken = () =>
+  fetch('http://localhost:4000/api/auth/token', {
+    credentials: 'include',
+  })
+    .then((response) => response.json())
+    .then((data) => data.access_token);
+
+const SigninPage = () => {
+  const { setAuth } = useAuth();
+  const navigate = useNavigate();
+  const [token] = createResource(fetchToken);
+  createEffect(() => {
+    if (!token()) {
+      return;
+    }
+
+    setAuth({ token: token() });
+    navigate(sessionStorage.getItem('auth_return_path'));
```

Comment:
what happens if auth_return_path undefined and i am not sure what's happening here after token is stored.

Reply:
Can't hurt to implement a fallback, I'll add it 👍 

The trick is to return the user to where they came from before going through the authentication loop.